src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -830,50 +839,98 @@ public LdapUserDetails updateUserDetails(LdapUserDetails d) {
 
     @Override
     public GroupDetails loadGroupByGroupname(String groupname) throws UsernameNotFoundException, DataAccessException {
+        return loadGroupByGroupname(groupname, false);
+    }
+
+    @Override
+    public GroupDetails loadGroupByGroupname(String groupname, boolean fetchMembers) throws UsernameNotFoundException, DataAccessException {
         groupname = fixGroupname(groupname);
-        Set<String> cachedGroups;
+        GroupDetails cachedGroup;
```

Comment:
It wasn't clear to me why the cache was a `Set` when the only use of the value was calling `.iterator().next()`. If I am missing something important please let me know and I can revert this part of the change.